clang/lib/Sema/SemaDeclCXX.cpp

```diff
@@ -18157,7 +18158,7 @@ void Sema::SetDeclDeleted(Decl *Dcl, SourceLocation DelLoc) {
   // C++11 [dcl.fct.def.delete]p4:
   //  A deleted function is implicitly inline.
   Fn->setImplicitlyInline();
-  Fn->setDeletedAsWritten();
+  Fn->setDeletedWithMessage(Message);
```

Comment:
I think we should just add the Message as parameter to `setDeletedAsWritten`

Reply:
I didn’t want to do that because you’d have to write `setDeletedAsWritten(true, Message)` everywhere, but honestly, that’s probably less confusing than having two ‘entirely different’ functions for this now that I think about it...

Reply:
> I think we should just add the Message as parameter to `setDeletedAsWritten`

I don’t think that would work too well after all now that the message is stored in the `DefaultFunctionInfo`, because `setDeletedAsWritten()` is called in many places, but the `DefaultFunctionInfo` needs to be constructed separately in some. Currently, there is only one place where `setDeletedAsWritten()` is called and then immediately after `setDeletedMessage()`, so I don’t thing it’s worth merging the two.